src/communication/network_interface_sr.c

```diff
@@ -10820,15 +10820,27 @@ sflashback_get_summary (THREAD_ENTRY * thread_p, unsigned int rid, char *request
   time_t start_time = 0;
   time_t end_time = 0;
 
-  if (flashback_min_log_pageid_to_keep () != NULL_LOG_PAGEID)
+  bool is_duplicated_request = false;
+
+  /* If multiple requests come at the same time,
+   * they can all be treated as non-duplicate requests.
+   * So, latch for check and set flashback connection is required */
+
+  flashback_lock_request ();
+
+  is_duplicated_request = flashback_is_duplicated_request (thread_p);
```

Comment:
Thanks for the detailed feedback. 
Lock is also required in checking if it is duplicated request.
If two connections are simultaneously enter the 'flashback_is_duplicated_request()' and 'flashback_Current_conn' is null, it is determined that both connections are not duplicated and both can proceed with initialization.

Reply:
> Lock is also required in checking if it is duplicated request

Yes, but it seems that the check is done only during initializing the flashback. Isn't it?

Reply:
Oh, now I understand what you mean. 
check and settings (initialization) are only done here, and are not shared anywhere else. 

Other mutex locks (css_Client_id_lock, css_Conn_rule_lock,.. ) are declared as static global variables, even if they are only used in specific function.

But I think static local variable is sufficient because it is only used in initialization functions. If it is used in other functions in the future, it would be better to modify it then.

Reply:
Yes, it looks better to me. My point was, by the way, that we don't have to open latch interfaces, but can just acquire and release the latch locally in the initialization function.

Reply:
Either the duplication check does.

Reply:
Yes, I agree with you